File: Day13/7_Sort_Stack.java
import java.util.* ;
import java.io.*; 
public class Solution {

	public static void sortStack(Stack<Integer> stack) {
		// Write your code here.
		sort(stack);
	}

	public static void sort(Stack<Integer> s) {
		if (!s.isEmpty()) {
			int x = s.pop();
			sort(s);
			insert(s, x);
		}
	}

	public static void insert(Stack<Integer> s, int x) {
		if (s.isEmpty() || x > s.peek()) {
			s.push(x);
			return;
		}

		int temp = s.pop();
		insert(s, x);
		s.push(temp);
	}

}

File: Day13/7_Sort_Stack.py
def insert(s, x):
    if not s or x > s[-1]:
        s.append(x)
        return

    temp = s.pop()
    insert(s, x)
    s.append(temp)

def sortStack(stack):
    # given data structure is a python list 
    # as list have all the similar operations available so use them
    # Write your code here
    if stack:
        x = stack.pop()
        sortStack(stack)
        insert(stack, x)
